### PR TITLE
Refactor reactpurecomponent

### PR DESCRIPTION
**Pull request by churchie317**
I'm curious to hear thoughts on this PR.

Looks to me like the code in this PR is shorter (smaller byte-size) and slightly more expressive. My personal opinion is that ComponentDummy parasitic inheritance pattern used here adds an unnecessary layer of misdirection. That said, if it exists for a good reason I would love to hear it!

The only criticism I might have of this PR is that Object.create is not very expressive if you're unfamiliar with its functionality.

### CODE REVIEW

File: src/isomorphic/children/__tests__/onlyChild-test.js [hunk 1]
```diff
@@ -94,4 +94,12 @@ describe('onlyChild', () => {
     }).not.toThrow();
   });
 
+  it('should return the only child', () => {
+    var instance =
```
Comment (gaearon) [inline src/isomorphic/children/__tests__/onlyChild-test.js]: Is this related to this PR?

Comment (churchie317) [inline src/isomorphic/children/__tests__/onlyChild-test.js]: Oh, whoops.
This is actually from an earlier PR that's still pending. Is there a good way to handle something like this?

Comment (gaearon) [inline src/isomorphic/children/__tests__/onlyChild-test.js]: Normally every time you want to start working on a PR you should pull the master branch from our repo and create a new local branch from it. You are likely seeing this because you forgot to switch back to master before creating the second feature branch.

Comment (gaearon) [inline src/isomorphic/children/__tests__/onlyChild-test.js]: You can fix this by running

```
git reset --hard origin/master
```

then cherry picking these two commits you want, and then force pushing.

**Comment by churchie317** [comment]
Fixed!

**Comment by gaearon** [comment]
I think the main reason might be IE8 compatibility.
We don't actually support it anymore but wouldn't be very nice to suddenly break existing users.

@sebmarkbage Any other reasons for indirection there?

**Comment by sebmarkbage** [comment]
The reason was because`Object.create` can't be faithfully shimmed. Taking a dependency on it means that people have to use "shams". I.e. non-faithful polyfills which some people don't want to do since their codebases may accidentally rely on non-standard or non-polyfillable behavior.

It was easy enough to get rid of it to avoid the controversy.

**Comment by churchie317** [comment]
Thanks a lot for the insight!

Closing because this makes total sense.
